app-modules/engagement/src/Notifications/EngagementEmailSentNotification.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Assist\Engagement\Notifications;
+
+use App\Models\User;
+use Illuminate\Bus\Queueable;
+use Assist\Engagement\Models\Engagement;
+use Illuminate\Notifications\Notification;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Notifications\Messages\MailMessage;
+use Filament\Notifications\Notification as FilamentNotification;
+
+class EngagementEmailSentNotification extends Notification implements ShouldQueue
+{
+    use Queueable;
+
+    public function __construct(
+        public Engagement $engagement
+    ) {}
+
+    public function via(User $notifiable): array
+    {
+        return ['mail', 'database'];
+    }
+
+    public function toMail(object $notifiable): MailMessage
+    {
+        return (new MailMessage())
+            ->subject('Your Engagement Email has successfully been delivered.')
+            // TODO Remove reliance on "preferred" as a field - extract to attribute
+            ->line("Your engagement was successfully delivered to {$this->engagement->recipient->preferred}.");
+    }
+
+    public function toDatabase(User $notifiable): array
+    {
+        return FilamentNotification::make()
+            ->success()
+            ->title('Engagement Email Successfully Delivered')
+            ->body("Your engagement email was successfully delivered to {$this->engagement->recipient->display_name}.")
```

Comment:
I noticed we use `display_name` here but in other areas we use `preferred` with a note to fix it later.

Should we just use display_name?

Reply:
The only difference here is in who receives the notification. I _think_ (though I could be wrong) we'll want to use `preferred` when engaging with the Educatable, whereas admins might want to see `display_name`. But, we can probably discuss this strategy and what the right approach is.

Reply:
Wouldn't both these notifications be going to an admin? Or do we we use toMail as like a response for an incoming Engagement?